# Data preparation

In [1]:
import pandas as pd
import plotly.express as px
import seaborn as sns
import plotly.graph_objects as go
import plotly.io as pio

In [2]:
import google.colab as gc

In [3]:
gc.drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


In [4]:
%cd '/content/drive/MyDrive/Solo learn'

/content/drive/MyDrive/Solo learn


In [5]:
!ls

'Fake news detection.ipynb'   open_one_time_covid_education_impact.csv
 Matplotlib.ipynb	     'Venezuela Covid19.ipynb'
 news.csv


# Reading data

In [6]:
df = pd.read_csv('open_one_time_covid_education_impact.csv')
df.head(10)

submission_id  ... are_children_promoted_with_a_modality_different_from_formal_evaluation
0  4570534489882624  ...                                                  0                    
1  6440166989496320  ...                                                  0                    
2  5001167808757760  ...                                                  1                    
3  5524933164924928  ...                                                  0                    
4  5032515751641088  ...                                                  1                    
5  5856891153678336  ...                                                  1                    
6  4994446973468672  ...                                                  0                    
7  6295201927921664  ...                                                  0                    
8  6354348190990336  ...                                                  1                    
9  5852928190709760  ...                                                  0                    

[10 rows x 27 columns]

In [7]:
print(df.shape)

(4436, 27)


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4436 entries, 0 to 4435
Data columns (total 27 columns):
 #   Column                                                                  Non-Null Count  Dtype 
---  ------                                                                  --------------  ----- 
 0   submission_id                                                           4436 non-null   int64 
 1   submission_date                                                         4436 non-null   object
 2   gender                                                                  4436 non-null   object
 3   age                                                                     4436 non-null   object
 4   geography                                                               4436 non-null   object
 5   financial_situation                                                     4436 non-null   object
 6   education                                                               4436 non-null   

In [9]:
df['submission_date'].unique()

array(['2021-03-17', '2021-03-29', '2021-03-18', '2021-03-24',
       '2021-03-15', '2021-03-31', '2021-03-14', '2021-03-23',
       '2021-03-30', '2021-03-16', '2021-02-07', '2021-04-07',
       '2021-03-21', '2021-03-13', '2021-03-28', '2021-06-10',
       '2021-03-22', '2021-04-03', '2021-03-19', '2021-04-01',
       '2021-03-26', '2021-03-20', '2021-03-25', '2021-04-02',
       '2021-03-27', '2021-05-22', '2021-04-04', '2021-05-01',
       '2021-04-16', '2021-04-09', '2021-06-22', '2021-05-13',
       '2021-04-13', '2021-04-15', '2021-04-12', '2021-04-05',
       '2021-05-23', '2021-05-03', '2021-06-16', '2021-05-19',
       '2021-05-05', '2021-04-06', '2021-05-25', '2021-06-27',
       '2021-06-05', '2021-04-19', '2021-05-18', '2021-05-06',
       '2021-04-20', '2021-04-26', '2021-05-24', '2021-04-08',
       '2021-05-04', '2021-06-07', '2021-06-01', '2021-03-12',
       '2021-06-24', '2021-04-24', '2021-04-23', '2021-05-12',
       '2021-06-02', '2021-05-14', '2021-03-11', '2021-

# data now

In [10]:
df.head(5)

submission_id  ... are_children_promoted_with_a_modality_different_from_formal_evaluation
0  4570534489882624  ...                                                  0                    
1  6440166989496320  ...                                                  0                    
2  5001167808757760  ...                                                  1                    
3  5524933164924928  ...                                                  0                    
4  5032515751641088  ...                                                  1                    

[5 rows x 27 columns]

In [11]:
df['age'].unique()

array(['Over 45 years old', '26 to 35 years old', '36 to 45 years old',
       '16 to 25 years old', 'Under 16', 'Not Available'], dtype=object)

In [12]:
df['age'].value_counts()

26 to 35 years old    1304
16 to 25 years old    1302
36 to 45 years old    1087
Over 45 years old      739
Not Available            3
Under 16                 1
Name: age, dtype: int64

In [13]:
df['gender'].unique()

array(['Female', 'Male', 'Prefer not to answer', 'Non-Binary',
       'Not Available'], dtype=object)

In [14]:
df['gender'].value_counts()

Female                  2341
Male                    2054
Prefer not to answer      35
Non-Binary                 5
Not Available              1
Name: gender, dtype: int64

In [15]:
df['submission_state'].value_counts()

Miranda             732
Zulia               611
Distrito Federal    556
Táchira             462
Lara                354
Anzoátegui          346
Bolívar             245
Aragua              169
Sucre               164
Carabobo            161
Mérida               91
Falcón               67
Monagas              66
Guárico              58
Portuguesa           54
Apure                54
Trujillo             53
Yaracuy              48
Vargas               46
Nueva Esparta        38
Barinas              32
Cojedes              18
Delta Amacuro         9
Amazonas              2
Name: submission_state, dtype: int64

In [16]:
df['geography'].unique()

array(['Suburban/Peri-urban', 'City center or metropolitan area', 'Rural',
       'Not Available'], dtype=object)

In [17]:
df['financial_situation'].unique()

array(['I can afford food and regular expenses, but nothing else',
       'I cannot afford enough food for my family',
       'I can comfortably afford food, clothes, and furniture, and I have savings',
       'I can afford food, but nothing else',
       'I can comfortably afford food, clothes, and furniture, but I don’t have savings',
       'I can afford food, regular expenses, and clothes, but nothing else',
       'Prefer not to answer', 'Not Available'], dtype=object)

In [18]:
df['education'].unique()

array(['University or college degree completed',
       'Technical school diploma or degree completed',
       'Some technical education (e.g polytechnic school)',
       'Some university or college',
       'Secondary school/ high school completed',
       'Primary school completed', 'Some primary education',
       'Some secondary school / high school', 'Post-graduate education',
       'Prefer not to answer', 'No formal education', 'Not available',
       'College or university', 'Post graduate', 'Technical school',
       'Secondary/high school'], dtype=object)

In [19]:
df['education'].value_counts()

University or college degree completed               1327
Secondary school/ high school completed              1089
Technical school diploma or degree completed          580
Some university or college                            379
Some technical education (e.g polytechnic school)     322
Some secondary school / high school                   279
Post-graduate education                               241
Primary school completed                               98
Prefer not to answer                                   58
Some primary education                                 36
No formal education                                    20
College or university                                   3
Secondary/high school                                   1
Not available                                           1
Technical school                                        1
Post graduate                                           1
Name: education, dtype: int64

'No formal education', 'Some primary education', 'Primary school completed',
                                    'Secondary/high school', 'Some secondary school / high school', 'Secondary school/ high school completed',
                                    'Technical school',  'Some technical education (e.g polytechnic school)', 'Technical school diploma or degree completed',
                                     'College or university', 'Some university or college', 'University or college degree completed',
                                    'Post graduate', 'Post-graduate education'

In [20]:
df['employment_status'].unique()

array(['I am unemployed',
       'I work full-time, either as an employee or self-employed',
       'I do housework, fulfilling domestic tasks, looking after children',
       'I work part-time, either as an employee or self-employed',
       'I am retired', 'I am a student and I work part-time',
       'I am a student', 'None of the above',
       'I am doing community or military service',
       'I am unable to work due to long-term illness or disability',
       'Not Available', 'Employed full-time', 'Retired', 'Student',
       'Unemployed'], dtype=object)

# check miss value

In [21]:
df[df.isnull().any(axis=1)]

Empty DataFrame
Columns: [submission_id, submission_date, gender, age, geography, financial_situation, education, employment_status, submission_state, are_there_children_0_to_2_yrs_out_of_educational_system, were_children_3_to_17_yrs_enrolled_and_did_not_return_to_school, are_there_children_who_stopped_enrolling_in_primary_education, are_there_children_who_stopped_enrolling_in_secondary_education, are_children_attending_face_to_face_classes, can_children_observe_deterioration_of_basic_services_of_school, do_children_3_and_17_yrs_receive_regular_school_meals, are_there_teachers_at_scheduled_class_hours, are_children_3_to_17_yrs_dealing_with_irregular_school_activity, are_children_being_teached_by_unqualified_people, did_teachers_leave_the_educational_system, do_school_and_the_teachers_have_internet_connection, do_children_have_internet_connection, do_children_3_to_17_yrs_miss_virtual_class_due_to_lack_of_electricity, does_home_shows_severe_deficit_of_electricity, does_home_shows_severe_deficit_of_internet, do_children_3_to_17_yrs_miss_class_or_in_lower_grade, are_children_promoted_with_a_modality_different_from_formal_evaluation]
Index: []

In [22]:
missing_values = ['na','--','n/a', 'Not Available']
df[df.isin(missing_values).any(axis=1)]

submission_id  ... are_children_promoted_with_a_modality_different_from_formal_evaluation
1167  6737975944937472  ...                                                  1                    
1352  6662974742986752  ...                                                  1                    
3343  5502504275279872  ...                                                  1                    

[3 rows x 27 columns]

In [23]:
# handle missing data - replace values when read data
missing_values = ['na','--','n/a', 'nan', 'Not Available']
df_replace_missing = pd.read_csv('open_one_time_covid_education_impact.csv',na_values=missing_values)
df_replace_missing

submission_id  ... are_children_promoted_with_a_modality_different_from_formal_evaluation
0     4570534489882624  ...                                                  0                    
1     6440166989496320  ...                                                  0                    
2     5001167808757760  ...                                                  1                    
3     5524933164924928  ...                                                  0                    
4     5032515751641088  ...                                                  1                    
...                ...  ...                                                ...                    
4431  5229259428986880  ...                                                  1                    
4432  6520660045332480  ...                                                  0                    
4433  5222701550796800  ...                                                  1                    
4434  6150639704080384  ...                                                  0                    
4435  5073215465717760  ...                                                  1                    

[4436 rows x 27 columns]

In [24]:
missing_values = ['na','--','n/a', 'nan', 'Not Available']
df_replace_missing[df_replace_missing.isin(missing_values).any(axis=1)]

Empty DataFrame
Columns: [submission_id, submission_date, gender, age, geography, financial_situation, education, employment_status, submission_state, are_there_children_0_to_2_yrs_out_of_educational_system, were_children_3_to_17_yrs_enrolled_and_did_not_return_to_school, are_there_children_who_stopped_enrolling_in_primary_education, are_there_children_who_stopped_enrolling_in_secondary_education, are_children_attending_face_to_face_classes, can_children_observe_deterioration_of_basic_services_of_school, do_children_3_and_17_yrs_receive_regular_school_meals, are_there_teachers_at_scheduled_class_hours, are_children_3_to_17_yrs_dealing_with_irregular_school_activity, are_children_being_teached_by_unqualified_people, did_teachers_leave_the_educational_system, do_school_and_the_teachers_have_internet_connection, do_children_have_internet_connection, do_children_3_to_17_yrs_miss_virtual_class_due_to_lack_of_electricity, does_home_shows_severe_deficit_of_electricity, does_home_shows_severe_deficit_of_internet, do_children_3_to_17_yrs_miss_class_or_in_lower_grade, are_children_promoted_with_a_modality_different_from_formal_evaluation]
Index: []

In [25]:
df = df_replace_missing.copy()
df.head()

submission_id  ... are_children_promoted_with_a_modality_different_from_formal_evaluation
0  4570534489882624  ...                                                  0                    
1  6440166989496320  ...                                                  0                    
2  5001167808757760  ...                                                  1                    
3  5524933164924928  ...                                                  0                    
4  5032515751641088  ...                                                  1                    

[5 rows x 27 columns]

In [26]:
missing_values = ['na','--','n/a', 'nan', 'Not Available']
df[df.isin(missing_values).any(axis=1)]

Empty DataFrame
Columns: [submission_id, submission_date, gender, age, geography, financial_situation, education, employment_status, submission_state, are_there_children_0_to_2_yrs_out_of_educational_system, were_children_3_to_17_yrs_enrolled_and_did_not_return_to_school, are_there_children_who_stopped_enrolling_in_primary_education, are_there_children_who_stopped_enrolling_in_secondary_education, are_children_attending_face_to_face_classes, can_children_observe_deterioration_of_basic_services_of_school, do_children_3_and_17_yrs_receive_regular_school_meals, are_there_teachers_at_scheduled_class_hours, are_children_3_to_17_yrs_dealing_with_irregular_school_activity, are_children_being_teached_by_unqualified_people, did_teachers_leave_the_educational_system, do_school_and_the_teachers_have_internet_connection, do_children_have_internet_connection, do_children_3_to_17_yrs_miss_virtual_class_due_to_lack_of_electricity, does_home_shows_severe_deficit_of_electricity, does_home_shows_severe_deficit_of_internet, do_children_3_to_17_yrs_miss_class_or_in_lower_grade, are_children_promoted_with_a_modality_different_from_formal_evaluation]
Index: []

# change columns name

In [27]:
df.columns

Index(['submission_id', 'submission_date', 'gender', 'age', 'geography',
       'financial_situation', 'education', 'employment_status',
       'submission_state',
       'are_there_children_0_to_2_yrs_out_of_educational_system',
       'were_children_3_to_17_yrs_enrolled_and_did_not_return_to_school',
       'are_there_children_who_stopped_enrolling_in_primary_education',
       'are_there_children_who_stopped_enrolling_in_secondary_education',
       'are_children_attending_face_to_face_classes',
       'can_children_observe_deterioration_of_basic_services_of_school',
       'do_children_3_and_17_yrs_receive_regular_school_meals',
       'are_there_teachers_at_scheduled_class_hours',
       'are_children_3_to_17_yrs_dealing_with_irregular_school_activity',
       'are_children_being_teached_by_unqualified_people',
       'did_teachers_leave_the_educational_system',
       'do_school_and_the_teachers_have_internet_connection',
       'do_children_have_internet_connection',
       'do_

In [28]:
# rename column name
df = df.rename(columns={'submission_date':'Date', 'geography':'region', 'financial_situation':'finance', 'employment_status':'employment', 'submission_state':'state',
                        'are_there_children_0_to_2_yrs_out_of_educational_system':'0_2enrolled_out',
                        'were_children_3_to_17_yrs_enrolled_and_did_not_return_to_school':'3_17enrolled_out',
                        'are_there_children_who_stopped_enrolling_in_primary_education':'CS_primary',
                        'are_there_children_who_stopped_enrolling_in_secondary_education':'CS_secondary',
                        'are_children_attending_face_to_face_classes':'C_F2F',
                        'can_children_observe_deterioration_of_basic_services_of_school':'deter',
                        'do_children_3_and_17_yrs_receive_regular_school_meals':'3_17meals',
                        'are_there_teachers_at_scheduled_class_hours':'t_scheduled',
                        'are_children_3_to_17_yrs_dealing_with_irregular_school_activity':'3_17irregular',
                        'are_children_being_teached_by_unqualified_people':'c_unquald',
                        'did_teachers_leave_the_educational_system':'t_leave',
                        'do_school_and_the_teachers_have_internet_connection':'st_internet',
                        'do_children_have_internet_connection':'c_internet',
                        'do_children_3_to_17_yrs_miss_virtual_class_due_to_lack_of_electricity':'3_17missvc',
                        'does_home_shows_severe_deficit_of_electricity':'deficit_electricity',
                        'does_home_shows_severe_deficit_of_internet':'deficit_internet',
                        'do_children_3_to_17_yrs_miss_class_or_in_lower_grade': '3_17misslg',
                        'are_children_promoted_with_a_modality_different_from_formal_evaluation':'c_promoted',
                        'submission_id':'id'})
df.columns

Index(['id', 'Date', 'gender', 'age', 'region', 'finance', 'education',
       'employment', 'state', '0_2enrolled_out', '3_17enrolled_out',
       'CS_primary', 'CS_secondary', 'C_F2F', 'deter', '3_17meals',
       't_scheduled', '3_17irregular', 'c_unquald', 't_leave', 'st_internet',
       'c_internet', '3_17missvc', 'deficit_electricity', 'deficit_internet',
       '3_17misslg', 'c_promoted'],
      dtype='object')

In [29]:
missing_values = ['na','--','n/a', 'nan', 'Not Available']
df[df.isin(missing_values).any(axis=1)]

Empty DataFrame
Columns: [id, Date, gender, age, region, finance, education, employment, state, 0_2enrolled_out, 3_17enrolled_out, CS_primary, CS_secondary, C_F2F, deter, 3_17meals, t_scheduled, 3_17irregular, c_unquald, t_leave, st_internet, c_internet, 3_17missvc, deficit_electricity, deficit_internet, 3_17misslg, c_promoted]
Index: []

# so now the data has clean from miss value.
1 = yes, 0 = no



In [30]:
df.head(3)

id        Date  ... 3_17misslg c_promoted
0  4570534489882624  2021-03-17  ...          0          0
1  6440166989496320  2021-03-29  ...          0          0
2  5001167808757760  2021-03-18  ...          0          1

[3 rows x 27 columns]

In [31]:
print(df.shape)

(4436, 27)


In [32]:
df['age'].value_counts()

26 to 35 years old    1304
16 to 25 years old    1302
36 to 45 years old    1087
Over 45 years old      739
Under 16                 1
Name: age, dtype: int64

# Overview

In [33]:
values = df['region'].value_counts()
labels = df['region'].value_counts().index

fig = px.pie(df, names=labels, values=values)
fig.update_layout(template='seaborn', hoverlabel=dict(font_size=20), title='Region Distribution')
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

In [34]:
fig = px.bar(
        df,
        orientation="h",
        title="Education distribution",
        x = df['education'].value_counts(),
        y = df['education'].value_counts().index,
        color_continuous_scale=["#2D6198", "#969696", "#737373", "#888889",],
        template='seaborn'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})

fig.show()

In [35]:
colors = ['lightslategray',] * 5

fig = px.bar(
        df,
        orientation="h",
        title="Finance",
        x = df['finance'].value_counts(),
        y = df['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [36]:
df['finance'].value_counts()

I can afford food, but nothing else                                                1445
I cannot afford enough food for my family                                          1163
I can afford food and regular expenses, but nothing else                           1060
I can afford food, regular expenses, and clothes, but nothing else                  244
Prefer not to answer                                                                239
I can comfortably afford food, clothes, and furniture, and I have savings           157
I can comfortably afford food, clothes, and furniture, but I don’t have savings     127
Name: finance, dtype: int64

In [37]:
fig = px.bar(
        df,
        orientation="h",
        title="Age",
        x = df['age'].value_counts(),
        y = df['age'].value_counts().index,
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})

fig.show()

In [38]:
df['age'].value_counts()

26 to 35 years old    1304
16 to 25 years old    1302
36 to 45 years old    1087
Over 45 years old      739
Under 16                 1
Name: age, dtype: int64

In [39]:
fig = px.bar(
        df,
        title="Submission State",
        x = df['state'].value_counts().index,
        y = df['state'].value_counts(),
        template='ggplot2'
        )

fig.update_layout(barmode='group', xaxis_tickangle=-45)

fig.show()

In [40]:
df['state'].value_counts()

Miranda             732
Zulia               611
Distrito Federal    556
Táchira             462
Lara                354
Anzoátegui          346
Bolívar             245
Aragua              169
Sucre               164
Carabobo            161
Mérida               91
Falcón               67
Monagas              66
Guárico              58
Portuguesa           54
Apure                54
Trujillo             53
Yaracuy              48
Vargas               46
Nueva Esparta        38
Barinas              32
Cojedes              18
Delta Amacuro         9
Amazonas              2
Name: state, dtype: int64

In [41]:
values = df['deficit_internet'].value_counts()
labels = df['deficit_internet'].value_counts().index

fig = px.pie(df, names=labels, values=values)
fig.update_layout(template='seaborn',
                  hoverlabel=dict(font_size=20),
                  title='Internet deficit at home?'
                  )
fig.update_traces(textposition='inside', textinfo='percent+value')
fig.show()

In [42]:
values = df['deficit_electricity'].value_counts()
labels = df['deficit_electricity'].value_counts().index

fig = px.pie(df, names=labels, values=values)
fig.update_layout(template='seaborn',
                  hoverlabel=dict(font_size=20),
                  title='Electric deficit at home?'
                  )
fig.update_traces(textposition='inside', textinfo='percent+value')
fig.show()

In [43]:
values = df['0_2enrolled_out'].value_counts()
labels = df['0_2enrolled_out'].value_counts().index

fig = px.pie(df, names=labels, values=values)
fig.update_layout(template='seaborn',
                  hoverlabel=dict(font_size=20),
                  title='Are there children 0-2 years out of educational system?'
                  )
fig.update_traces(textposition='inside', textinfo='percent+value')
fig.show()

In [44]:
values = df['3_17enrolled_out'].value_counts()
labels = df['3_17enrolled_out'].value_counts().index

fig = px.pie(df, names=labels, values=values)
fig.update_layout(template='seaborn',
                  hoverlabel=dict(font_size=20),
                  title="Were children 3-17 years enrolled and didn't returned to school?"
                  )
fig.update_traces(textposition='inside', textinfo='percent+value')
fig.show()

In [45]:
values = df['deter'].value_counts()
labels = df['deter'].value_counts().index

fig = px.pie(df, names=labels, values=values)
fig.update_layout(template='seaborn',
                  hoverlabel=dict(font_size=20),
                  title='Do children observe deteration of basic school service?'
                  )
fig.update_traces(textposition='inside', textinfo='percent+value')
fig.show()

In [46]:
values = df['st_internet'].value_counts()
labels = df['st_internet'].value_counts().index

fig = px.pie(df, names=labels, values=values)
fig.update_layout(template='seaborn',
                  hoverlabel=dict(font_size=20),
                  title='Does school and teacher has internet?'
                  )
fig.update_traces(textposition='inside', textinfo='percent+value')
fig.show()

In [47]:
values = df['3_17missvc'].value_counts()
labels = df['3_17missvc'].value_counts().index

fig = px.pie(df, names=labels, values=values)
fig.update_layout(template='seaborn',
                  hoverlabel=dict(font_size=20),
                  title='Does children 3-17 years old miss class because lack of electricity?'
                  )
fig.update_traces(textposition='inside', textinfo='percent+value')
fig.show()

# We'll classified the data into several age categories, 16-25 (teen) and 26+ (adults)
Only 4433 data records because 3 other are Not Available

# Age centric data


## Age 0-25 (teen)

In [48]:
df1625 = df[df['age'].isin(['Under 16', '16 to 25 years old'])]

In [49]:
print(df1625.shape)

(1303, 27)


In [50]:
values = df1625['region'].value_counts()
labels = df1625['region'].value_counts().index

fig = px.pie(df1625, names=labels, values=values, color_discrete_sequence=['green', 'pink'])
fig.update_layout(title='Region Distribution in Age 16-25', template='plotly_dark', hoverlabel=dict(font_size=20))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

In [51]:
values = df1625['education'].value_counts()
labels = df1625['education'].value_counts().index

fig = px.bar(
        df1625,
        orientation="h",
        title="Education in 16-25",
        x = df1625['education'].value_counts(),
        y = df1625['education'].value_counts().index,
        color_continuous_scale=["#bdbdbd", "#969696", "#737373", "#525252",],
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})

fig.show()

In [52]:
fig = px.bar(df1625, orientation="h", title="Employment in Age 16-25",
             x=df1625['employment'].value_counts(),
             y=df1625['employment'].value_counts().index,
             )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})

fig.show()

In [53]:
fig = px.bar(
        df1625,
        orientation="h",
        title="Financial condition in 16-25",
        x = df1625['finance'].value_counts(),
        y = df1625['finance'].value_counts().index,
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})

fig.show()

## Age 26++ (adults)

In [54]:
df26 = df[df['age'].isin(['26 to 35 years old', '36 to 45 years old', 'Over 45 years old'])]

In [55]:
print(df26.shape)

(3130, 27)


In [56]:
fig = px.bar(
        df26,
        orientation="h",
        title="Education in 26+",
        x = df26['education'].value_counts(),
        y = df26['education'].value_counts().index,
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})

fig.show()

In [57]:
fig = px.bar(df26, 
             orientation="h", 
             title="Employment in Age 26+",
             x=df26['employment'].value_counts(),
             y=df26['employment'].value_counts().index,
             )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})

fig.show()

In [58]:
fig = px.bar(
        df26,
        orientation="h",
        title="Financial condition in 26+",
        x = df26['finance'].value_counts(),
        y = df26['finance'].value_counts().index,
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})

fig.show()

# Per state view

We'll look  into several columns ONLY.
Internet,
electricity,
3-17 enrolled and didn't back to school,
deteriation of school service,
school/teacher internet availability,
3-17 years miss class bcs lack of electricity,

In [59]:
dfnew = df[['id', 'Date', 'gender', 'age', 'region', 'finance', 'education',
       'employment', 'state', 'deficit_internet', 'deficit_electricity', '3_17enrolled_out', 'deter', 'st_internet',
       '3_17missvc']]
dfnew.columns

Index(['id', 'Date', 'gender', 'age', 'region', 'finance', 'education',
       'employment', 'state', 'deficit_internet', 'deficit_electricity',
       '3_17enrolled_out', 'deter', 'st_internet', '3_17missvc'],
      dtype='object')

## State Miranda

In [60]:
dfmiranda = dfnew[dfnew['state'].isin(['Miranda'])]
print(dfmiranda.shape)

(732, 15)


In [61]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0.2, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Miranda as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [62]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfmiranda,
        orientation="h",
        title="Finance",
        x = dfmiranda['finance'].value_counts(),
        y = dfmiranda['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [63]:
dfmiranda['finance'].value_counts()

I can afford food, but nothing else                                                240
I can afford food and regular expenses, but nothing else                           205
I cannot afford enough food for my family                                          172
I can afford food, regular expenses, and clothes, but nothing else                  41
Prefer not to answer                                                                38
I can comfortably afford food, clothes, and furniture, and I have savings           19
I can comfortably afford food, clothes, and furniture, but I don’t have savings     17
Name: finance, dtype: int64

In [64]:
dfmiranda['age'].value_counts()

26 to 35 years old    216
36 to 45 years old    213
16 to 25 years old    170
Over 45 years old     133
Name: age, dtype: int64

#### Miranda, deficit electric & internet, and enrolled out of school (Student view)

In [65]:
filter_a = dfmiranda['deficit_internet'] > 0
filter_b = dfmiranda['deficit_electricity'] > 0
dfmirinel = dfmiranda[(filter_a & filter_b)]
print(dfmirinel.shape)

(128, 15)


In [66]:
filter_z = dfmirinel['3_17enrolled_out'] < 1
dfmirinel_317 = dfmirinel[filter_z]
print(dfmirinel_317.shape)

(35, 15)


In [67]:
filter_c = dfmirinel['3_17enrolled_out'] > 0
dfmirinel317 = dfmirinel[(filter_c)]
print(dfmirinel317.shape)

(93, 15)


In [68]:
dfmirinel_317['id'].unique()

array([5941509894701056, 5216724835368960, 5819398672875520,
       4967010470723584, 4601327811559424, 5461194977837056,
       4536312190468096, 6627455430557696, 5675878045712384,
       5578966269427712, 6431462449741824, 4814534970966016,
       5542302130110464, 6207126107324416, 5677247469191168,
       5939685540233216, 5443204618125312, 5050061917782016,
       4794629928255488, 6049440745127936, 6547997931601920,
       6595643480211456, 6703143089078272, 6727403010523136,
       5181253052792832, 4894572827377664, 6352958483857408,
       5422052709498880, 6394715045101568, 5888490066935808,
       5028090609926144, 5023307945738240, 6391028151222272,
       5112900175003648, 5967262451499008])

In [69]:
dfmirinel317['id'].unique()

array([5775884077498368, 6709223286374400, 6380540109520896,
       5647462894403584, 4532593117888512, 5911435258626048,
       5172460164218880, 4721531279376384, 5376410511474688,
       5258987984650240, 5131950313701376, 5488081473110016,
       5621009570070528, 6261625215778816, 6132062225694720,
       4877509509513216, 4824319174901760, 6550836837875712,
       6477638649774080, 5019539195822080, 6271055554674688,
       4924606392827904, 6250165907488768, 6035053913309184,
       6616740292460544, 5245209058476032, 5839327908331520,
       6710136285691904, 6194713077219328, 4552669925736448,
       6644681202204672, 6469479470202880, 5484528041984000,
       6559969683177472, 6070627734913024, 5598774457270272,
       4872766070456320, 4661098405167104, 4809200403480576,
       5835568620765184, 5497671799537664, 4560987012464640,
       5020663135076352, 5725103605415936, 5281885931700224,
       6061080442044416, 5530491590803456, 6368299184291840,
       4878839221321728,

#### Miranda, school/teachers internet availability (School view)

In [70]:
values = dfmiranda['st_internet'].value_counts()
labels = dfmiranda['st_internet'].value_counts().index

fig = px.pie(dfmiranda, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Zulia

In [71]:
dfzulia = dfnew[dfnew['state'].isin(['Zulia'])]
print(dfzulia.shape)

(611, 15)


In [72]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0.3, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Zulia as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [73]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfzulia,
        orientation="h",
        title="Finance",
        x = dfzulia['finance'].value_counts(),
        y = dfzulia['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [74]:
dfzulia['finance'].value_counts()

I can afford food, but nothing else                                                238
I cannot afford enough food for my family                                          178
I can afford food and regular expenses, but nothing else                           116
Prefer not to answer                                                                39
I can comfortably afford food, clothes, and furniture, but I don’t have savings     18
I can comfortably afford food, clothes, and furniture, and I have savings           12
I can afford food, regular expenses, and clothes, but nothing else                  10
Name: finance, dtype: int64

In [75]:
dfzulia['age'].value_counts()

36 to 45 years old    172
16 to 25 years old    168
26 to 35 years old    166
Over 45 years old     105
Name: age, dtype: int64

#### Zulia, deficit electric & internet, and enrolled out of school (Student view)

In [76]:
filter_a = dfzulia['deficit_internet'] > 0
filter_b = dfzulia['deficit_electricity'] > 0
dfzulinel = dfzulia[(filter_a & filter_b)]
print(dfzulinel.shape)

(170, 15)


In [77]:
filter_y = dfzulinel['3_17enrolled_out'] < 1
dfzulinel_317 = dfzulinel[(filter_y)]
print(dfzulinel_317.shape)

(42, 15)


In [78]:
filter_c = dfzulinel['3_17enrolled_out'] > 0
dfzulinel317 = dfzulinel[(filter_c)]
print(dfzulinel317.shape)

(128, 15)


In [79]:
dfzulinel_317['id'].unique()

array([4743733626273792, 5106413264437248, 6080045071466496,
       5146814981603328, 4586320457629696, 5922224333914112,
       4922605474938880, 5046899144130560, 4913721167314944,
       6631000707956736, 6573102619426816, 5443756739526656,
       6557997219446784, 5780293264080896, 6222914474213376,
       5119728770351104, 5665260886556672, 5214623522619392,
       5824548942643200, 5710415052144640, 5947298705309696,
       6497998472478720, 5923219910688768, 4856290240364544,
       5249589656682496, 4851575356129280, 5150290952323072,
       4676143742713856, 6634416280698880, 5104079989899264,
       5937744617668608, 6126067122438144, 6571571396476928,
       5257449547497472, 5410156254068736, 4935368691679232,
       5147541502164992, 6622436828381184, 6317703060258816,
       4956913673699328, 4739158831792128, 5689304751013888])

In [80]:
dfzulinel317['id'].unique()

array([6753172445200384, 6215360297828352, 5551783756955648,
       6410743527505920, 6010102267248640, 5874022704480256,
       6527940065230848, 6310425087180800, 5332048331145216,
       5501054757371904, 5771830131687424, 6433511149142016,
       6124152892424192, 6084617466806272, 6562433484455936,
       6727208260599808, 4835719729381376, 6054683692236800,
       5366575707455488, 4704834124251136, 5182275087237120,
       6579156241612800, 4697450135945216, 6662254782316544,
       5230646032334848, 5133647014526976, 6205665399013376,
       5099409145856000, 5241455122841600, 5348248545132544,
       6617056643645440, 5858766980644864, 6743370423861248,
       5803228322332672, 6281025784381440, 4549346510241792,
       5745950101602304, 5260479277826048, 6715079893712896,
       5345004234211328, 5119707681390592, 5749432179228672,
       5319140343808000, 5058596353343488, 5679628726566912,
       6597191010615296, 6488283726217216, 4792696874205184,
       5403746418032640,

#### Zulia, school/teachers internet availability (Teacher view)

In [81]:
values = dfzulia['st_internet'].value_counts()
labels = dfzulia['st_internet'].value_counts().index

fig = px.pie(dfzulia, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Distritio Federal

In [82]:
dfdf = df[df['state'].isin(['Distrito Federal'])]
print(dfdf.shape)

(556, 27)


In [83]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0.3, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Distrito Federal as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [84]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfdf,
        orientation="h",
        title="Finance",
        x = dfdf['finance'].value_counts(),
        y = dfdf['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [85]:
dfdf['finance'].value_counts()

I can afford food, but nothing else                                                170
I can afford food and regular expenses, but nothing else                           159
I cannot afford enough food for my family                                          128
I can afford food, regular expenses, and clothes, but nothing else                  35
Prefer not to answer                                                                34
I can comfortably afford food, clothes, and furniture, and I have savings           16
I can comfortably afford food, clothes, and furniture, but I don’t have savings     14
Name: finance, dtype: int64

In [86]:
dfdf['age'].value_counts()

16 to 25 years old    159
26 to 35 years old    158
36 to 45 years old    136
Over 45 years old     103
Name: age, dtype: int64

#### Distrito Federal, deficit electric & internet, and enrolled out of school (Student view)

In [87]:
filter_a = dfdf['deficit_internet'] > 0
filter_b = dfdf['deficit_electricity'] > 0
dfdfinel = dfdf[(filter_a & filter_b)]
print(dfdfinel.shape)

(86, 27)


In [88]:
filter_x = dfdfinel['3_17enrolled_out'] < 1
dfdfinel_317 = dfdfinel[(filter_x)]
print(dfdfinel_317.shape)

(29, 27)


In [89]:
filter_c = dfdfinel['3_17enrolled_out'] > 0
dfdfinel317 = dfdfinel[(filter_c)]
print(dfdfinel317.shape)

(57, 27)


In [90]:
dfdfinel_317['id'].unique()

array([5282975309889536, 5017936300605440, 6625890116304896,
       4734153047408640, 5261891315433472, 6532010050060288,
       6564731728166912, 4850278225674240, 5935718533955584,
       4548336656056320, 6660213330673664, 5917830263603200,
       5229644096995328, 4723955251544064, 6355843376480256,
       4850385918623744, 5875833653952512, 5255493760319488,
       5336027417018368, 5658764194873344, 5003597267075072,
       5176054045212672, 5486648497537024, 4842312856502272,
       5644600281923584, 5463787930124288, 5965437694377984,
       6486867863404544, 6228772809605120])

In [91]:
dfdfinel317['id'].unique()

array([5229614954971136, 6270746233143296, 5484427462574080,
       4868180253343744, 4977143875633152, 4840858624196608,
       4987973937659904, 5582169979486208, 5195786853023744,
       6136426633887744, 5286182442500096, 4655028408418304,
       5705981739925504, 5213375364857856, 5921561633882112,
       4506388784218112, 6059441576476672, 4687495945256960,
       6695057192386560, 5833891821912064, 6256979168460800,
       6103012140646400, 6601129965387776, 5212772442046464,
       5231685481529344, 5204878443151360, 6141727042961408,
       5751881061367808, 5748346760462336, 4585423899983872,
       6189345009827840, 4617931635032064, 5467146661658624,
       4767749825101824, 6703075762110464, 5869765083267072,
       4871146398023680, 4747417584140288, 4727058164088832,
       6256862835245056, 6581136422862848, 6385053985931264,
       4690179762946048, 6616285478912000, 6118016474677248,
       5294508890914816, 6477997078216704, 5187693943455744,
       5579249066180608,

#### Distrito Federal, school/teachers internet availability (School view)

In [92]:
values = dfdf['st_internet'].value_counts()
labels = dfdf['st_internet'].value_counts().index

fig = px.pie(dfdf, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Táchira

In [93]:
dfta = df[df['state'].isin(['Táchira'])]
print(dfta.shape)

(462, 27)


In [94]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0.3, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Táchira as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [95]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfta,
        orientation="h",
        title="Finance",
        x = dfta['finance'].value_counts(),
        y = dfta['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [96]:
dfta['finance'].value_counts()

I can afford food and regular expenses, but nothing else                           144
I can afford food, but nothing else                                                142
I cannot afford enough food for my family                                           92
I can afford food, regular expenses, and clothes, but nothing else                  35
Prefer not to answer                                                                28
I can comfortably afford food, clothes, and furniture, and I have savings           12
I can comfortably afford food, clothes, and furniture, but I don’t have savings      8
Name: finance, dtype: int64

In [97]:
dfta['age'].value_counts()

16 to 25 years old    134
26 to 35 years old    130
36 to 45 years old    101
Over 45 years old      96
Name: age, dtype: int64

#### Táchira, deficit electric & internet, and enrolled out of school (Student view)

In [98]:
filter_a = dfta['deficit_internet'] > 0
filter_b = dfta['deficit_electricity'] > 0
dftainel = dfta[(filter_a & filter_b)]
print(dftainel.shape)

(169, 27)


In [99]:
filter_w = dftainel['3_17enrolled_out'] < 1
dftainel_317 = dftainel[(filter_w)]
print(dftainel_317.shape)

(63, 27)


In [100]:
filter_c = dftainel['3_17enrolled_out'] > 0
dftainel317 = dftainel[(filter_c)]
print(dftainel317.shape)

(106, 27)


In [101]:
dftainel_317['id'].unique()

array([5045689775292416, 6652890629537792, 4624955114520576,
       5086421567406080, 6366402620686336, 5995557628936192,
       4969205131902976, 5874281644032000, 6420882603114496,
       5201073974542336, 5040730329579520, 5603410069159936,
       4718225832280064, 4951794257231872, 6022528211615744,
       5485748836106240, 6533556842266624, 6646557079502848,
       4794706986008576, 6242633977692160, 4772720427663360,
       5686726663340032, 6232950298640384, 4623864960385024,
       5200953715458048, 5552666087522304, 6211807353241600,
       5255464232419328, 5606204683583488, 5779496279212032,
       6460782849294336, 4664863950635008, 4775220853604352,
       4985930808295424, 6588523766611968, 5172688904781824,
       4850472421949440, 6712949036285952, 6532669126213632,
       6381472167755776, 5181589318533120, 4615767642603520,
       5893756837691392, 5970397291085824, 5520670812536832,
       5701225130754048, 4997892879417344, 4830143553208320,
       6554950477021184,

In [102]:
dftainel317['id'].unique()

array([5348441885769728, 4545290047848448, 5838656836468736,
       6014815641534464, 4689682284937216, 6082126167670784,
       5610948609179648, 4928072364326912, 4838613765586944,
       5752231390609408, 6345274133053440, 4954320117432320,
       4698412795822080, 4684529934794752, 5919861514698752,
       6728988457435136, 5918730126360576, 5861652158480384,
       6126883635986432, 5051423825068032, 6625261423689728,
       6640140683575296, 5572369199661056, 6001093841780736,
       5520344059478016, 5895504553836544, 5221989928402944,
       5282481455759360, 5882960900521984, 6281893686542336,
       5720876149637120, 5560958545297408, 6401762885107712,
       4720058508247040, 5469113689571328, 5898530626732032,
       4843795375194112, 5291061525348352, 4573278068015104,
       4533828122312704, 6283971393748992, 6433007430008832,
       6003237902876672, 6309007294988288, 6288089730514944,
       6556630278995968, 5464743946223616, 5098638450884608,
       6390189357531136,

#### Táchira, school/teachers internet availability (School view)

In [103]:
values = dfta['st_internet'].value_counts()
labels = dfta['st_internet'].value_counts().index

fig = px.pie(dfta, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Lara

In [104]:
dfla = df[df['state'].isin(['Lara'])]
print(dfla.shape)

(354, 27)


In [105]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0.3,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Lara as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [106]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfla,
        orientation="h",
        title="Finance",
        x = dfla['finance'].value_counts(),
        y = dfla['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [107]:
dfla['finance'].value_counts()

I can afford food, but nothing else                                                113
I cannot afford enough food for my family                                           94
I can afford food and regular expenses, but nothing else                            82
I can afford food, regular expenses, and clothes, but nothing else                  21
I can comfortably afford food, clothes, and furniture, and I have savings           18
Prefer not to answer                                                                15
I can comfortably afford food, clothes, and furniture, but I don’t have savings     11
Name: finance, dtype: int64

In [108]:
dfla['age'].value_counts()

16 to 25 years old    107
26 to 35 years old    101
36 to 45 years old     85
Over 45 years old      61
Name: age, dtype: int64

#### Lara, deficit electric & internet, and enrolled out of school (Student view)

In [109]:
filter_a = dfla['deficit_internet'] > 0
filter_b = dfla['deficit_electricity'] > 0
dflainel = dfla[(filter_a & filter_b)]
print(dflainel.shape)

(92, 27)


In [110]:
filter_v = dflainel['3_17enrolled_out'] < 1
dflainel_317 = dflainel[(filter_v)]
print(dflainel_317.shape)

(23, 27)


In [111]:
filter_c = dflainel['3_17enrolled_out'] > 0
dflainel317 = dflainel[(filter_c)]
print(dflainel317.shape)

(69, 27)


In [112]:
dflainel_317['id'].unique()

array([5116325461753856, 4563271666630656, 6201466481278976,
       6438489016238080, 5625022344593408, 5461245309485056,
       6165366542958592, 4825980857483264, 4781893219516416,
       5374154781229056, 5423389534846976, 5228724101578752,
       5253731615768576, 6309801578725376, 6124102896320512,
       5694988066488320, 5424240794009600, 6343931150467072,
       5415091658031104, 6444256922435584, 6031669781069824,
       6153550819491840, 5716877400866816])

In [113]:
dftainel317['id'].unique()

array([5348441885769728, 4545290047848448, 5838656836468736,
       6014815641534464, 4689682284937216, 6082126167670784,
       5610948609179648, 4928072364326912, 4838613765586944,
       5752231390609408, 6345274133053440, 4954320117432320,
       4698412795822080, 4684529934794752, 5919861514698752,
       6728988457435136, 5918730126360576, 5861652158480384,
       6126883635986432, 5051423825068032, 6625261423689728,
       6640140683575296, 5572369199661056, 6001093841780736,
       5520344059478016, 5895504553836544, 5221989928402944,
       5282481455759360, 5882960900521984, 6281893686542336,
       5720876149637120, 5560958545297408, 6401762885107712,
       4720058508247040, 5469113689571328, 5898530626732032,
       4843795375194112, 5291061525348352, 4573278068015104,
       4533828122312704, 6283971393748992, 6433007430008832,
       6003237902876672, 6309007294988288, 6288089730514944,
       6556630278995968, 5464743946223616, 5098638450884608,
       6390189357531136,

#### Lara, school/teachers internet availability (School view)

In [114]:
values = dfla['st_internet'].value_counts()
labels = dfla['st_internet'].value_counts().index

fig = px.pie(dfla, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Anzoátegui

In [115]:
dfan = df[df['state'].isin(['Anzoátegui'])]
print(dfan.shape)

(346, 27)


In [116]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0.3, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Anzoátegui as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [117]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfan,
        orientation="h",
        title="Finance",
        x = dfan['finance'].value_counts(),
        y = dfan['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [118]:
dfan['finance'].value_counts()

I can afford food, but nothing else                                                122
I cannot afford enough food for my family                                           94
I can afford food and regular expenses, but nothing else                            64
I can comfortably afford food, clothes, and furniture, and I have savings           20
I can afford food, regular expenses, and clothes, but nothing else                  17
Prefer not to answer                                                                15
I can comfortably afford food, clothes, and furniture, but I don’t have savings     14
Name: finance, dtype: int64

In [119]:
dfan['age'].value_counts()

26 to 35 years old    130
16 to 25 years old     96
36 to 45 years old     72
Over 45 years old      46
Under 16                1
Name: age, dtype: int64

#### Anzoátegui, deficit electric & internet, and enrolled out of school (Student view)

In [120]:
filter_a = dfan['deficit_internet'] > 0
filter_b = dfan['deficit_electricity'] > 0
dfaninel = dfan[(filter_a & filter_b)]
print(dfaninel.shape)

(51, 27)


In [121]:
filter_u = dfaninel['3_17enrolled_out'] < 1
dfaninel_317 = dfaninel[(filter_u)]
print(dfaninel_317.shape)

(15, 27)


In [122]:
filter_c = dfaninel['3_17enrolled_out'] > 0
dfaninel317 = dfaninel[(filter_c)]
print(dfaninel317.shape)

(36, 27)


In [123]:
dfaninel_317['id'].unique()

array([6001157393874944, 6750317533462528, 6366033219944448,
       6244345488605184, 5628249257541632, 6315273450946560,
       5750516088045568, 4528166818545664, 6070310192545792,
       6025758027022336, 5356663040376832, 5778704797270016,
       4665246068506624, 4986871439687680, 5073215465717760])

In [124]:
dfaninel317['id'].unique()

array([5112890075119616, 6454895858417664, 5691644535111680,
       4917640224309248, 5581298621218816, 4760136643248128,
       6287267361718272, 4655357980049408, 4976863729680384,
       6069211687550976, 5126988338823168, 5576968304328704,
       5724462380220416, 6302791940440064, 4514972679012352,
       4614922440015872, 6446197710782464, 5824562347638784,
       6138407637221376, 5125140546584576, 5354301831774208,
       6655627731664896, 4924043550785536, 5506984362377216,
       4729439589236736, 4693664390709248, 6393069737082880,
       6421582817001472, 6496843461820416, 6627208335720448,
       6162457306595328, 5238665994567680, 5877909884764160,
       4749287035764736, 4887661621018624, 5892247139909632])

#### Anzoátegui, school/teachers internet availability (School view)

In [125]:
values = dfan['st_internet'].value_counts()
labels = dfan['st_internet'].value_counts().index

fig = px.pie(dfan, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Bolívar

In [126]:
dfbo = df[df['state'].isin(['Bolívar'])]
print(dfbo.shape)

(245, 27)


In [127]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0.3, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Bolívar as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [128]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfbo,
        orientation="h",
        title="Finance",
        x = dfbo['finance'].value_counts(),
        y = dfbo['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [129]:
dfbo['finance'].value_counts()

I can afford food, but nothing else                                                87
I cannot afford enough food for my family                                          68
I can afford food and regular expenses, but nothing else                           46
I can afford food, regular expenses, and clothes, but nothing else                 18
I can comfortably afford food, clothes, and furniture, and I have savings          11
Prefer not to answer                                                                8
I can comfortably afford food, clothes, and furniture, but I don’t have savings     7
Name: finance, dtype: int64

In [130]:
dfbo['age'].value_counts()

16 to 25 years old    84
26 to 35 years old    78
36 to 45 years old    43
Over 45 years old     40
Name: age, dtype: int64

#### Bolívar, deficit electric & internet, and enrolled out of school (Student view)

In [131]:
filter_a = dfbo['deficit_internet'] > 0
filter_b = dfbo['deficit_electricity'] > 0
dfboinel = dfbo[(filter_a & filter_b)]
print(dfboinel.shape)

(47, 27)


In [132]:
filter_t = dfboinel['3_17enrolled_out'] < 1
dfboinel_317 = dfboinel[(filter_t)]
print(dfboinel_317.shape)

(13, 27)


In [133]:
filter_c = dfboinel['3_17enrolled_out'] > 0
dfboinel317 = dfboinel[(filter_c)]
print(dfboinel317.shape)

(34, 27)


In [134]:
dfboinel_317['id'].unique()

array([5747873911406592, 6529061336907776, 4952621390757888,
       5358509406552064, 6596118141206528, 6330477366738944,
       4758676488585216, 6290255467773952, 5924964674043904,
       6025294438989824, 5179152461463552, 6055606304899072,
       5942251296653312])

In [135]:
dfboinel317['id'].unique()

array([5781548065619968, 5865641914662912, 6252577045086208,
       5070098191876096, 6371572217806848, 5981979895136256,
       4606112891666432, 6438333726326784, 6288629151563776,
       6613785405292544, 6084304303292416, 5131324309635072,
       4908300281053184, 5376311643340800, 5977263014412288,
       6675294521131008, 5921300597178368, 5416292235608064,
       6281297508171776, 4695780266082304, 5242809379717120,
       5756104008269824, 5806480283664384, 5286590900600832,
       5817483251679232, 4751718507347968, 5792838259572736,
       5014461772726272, 5064728492900352, 5866002624806912,
       5689900442845184, 6603507481444352, 6441619762184192,
       4977344229146624])

#### Bolívar, school/teachers internet availability (School view)

In [136]:
values = dfbo['st_internet'].value_counts()
labels = dfbo['st_internet'].value_counts().index

fig = px.pie(dfbo, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Aragua

In [137]:
dfar = df[df['state'].isin(['Aragua'])]
print(dfar.shape)

(169, 27)


In [138]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0.3, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Aragua as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [139]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfar,
        orientation="h",
        title="Finance",
        x = dfar['finance'].value_counts(),
        y = dfar['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [140]:
dfar['finance'].value_counts()

I cannot afford enough food for my family                                          51
I can afford food, but nothing else                                                48
I can afford food and regular expenses, but nothing else                           37
Prefer not to answer                                                               11
I can afford food, regular expenses, and clothes, but nothing else                  9
I can comfortably afford food, clothes, and furniture, but I don’t have savings     7
I can comfortably afford food, clothes, and furniture, and I have savings           6
Name: finance, dtype: int64

In [141]:
dfar['age'].value_counts()

16 to 25 years old    52
36 to 45 years old    44
26 to 35 years old    38
Over 45 years old     34
Name: age, dtype: int64

#### Aragua, deficit electric & internet, and enrolled out of school (Student view)

In [142]:
filter_a = dfar['deficit_internet'] > 0
filter_b = dfar['deficit_electricity'] > 0
dfarinel = dfar[(filter_a & filter_b)]
print(dfarinel.shape)

(46, 27)


In [143]:
filter_s = dfarinel['3_17enrolled_out'] < 1
dfarinel_317 = dfarinel[(filter_s)]
print(dfarinel_317.shape)

(13, 27)


In [144]:
filter_c = dfarinel['3_17enrolled_out'] > 0
dfarinel317 = dfarinel[(filter_c)]
print(dfarinel317.shape)

(33, 27)


In [145]:
dfarinel_317['id'].unique()

array([6057277147840512, 4754248662515712, 4829466256998400,
       6604622780432384, 5392972509347840, 6454789675417600,
       5868550865813504, 5185006753546240, 4706631299629056,
       5664071381942272, 4723712736886784, 6132666876559360,
       5467637495889920])

In [146]:
dfarinel317['id'].unique()

array([4717614067875840, 6549568413237248, 5502504275279872,
       5651032532320256, 6583254412165120, 5874363852390400,
       5210453293924352, 5480308857372672, 5154044804071424,
       6027168655343616, 5254289340760064, 5302638207041536,
       5365331324567552, 5485758214569984, 4622645697183744,
       6418986240835584, 6510251410194432, 6435936664813568,
       5705325213908992, 6010504702328832, 5198463825608704,
       4910597417467904, 5357399136534528, 4656806105448448,
       6187429521522688, 4553015687380992, 6148170081107968,
       4712499701350400, 4645499717287936, 6008305410637824,
       5219684655038464, 6259929844547584, 5137382159679488])

#### Aragua, school/teachers internet availability (School view)

In [147]:
values = dfar['st_internet'].value_counts()
labels = dfar['st_internet'].value_counts().index

fig = px.pie(dfar, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Sucre

In [148]:
dfsu = df[df['state'].isin(['Sucre'])]
print(dfsu.shape)

(164, 27)


In [149]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0.5, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Sucre as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [150]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfsu,
        orientation="h",
        title="Finance",
        x = dfsu['finance'].value_counts(),
        y = dfsu['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [151]:
dfsu['finance'].value_counts()

I cannot afford enough food for my family                                          60
I can afford food, but nothing else                                                49
I can afford food and regular expenses, but nothing else                           33
I can afford food, regular expenses, and clothes, but nothing else                  7
Prefer not to answer                                                                6
I can comfortably afford food, clothes, and furniture, and I have savings           5
I can comfortably afford food, clothes, and furniture, but I don’t have savings     4
Name: finance, dtype: int64

In [152]:
dfsu['age'].value_counts()

26 to 35 years old    55
16 to 25 years old    55
36 to 45 years old    29
Over 45 years old     25
Name: age, dtype: int64

#### Sucre, deficit electric & internet, and enrolled out of school (Student view)

In [153]:
filter_a = dfsu['deficit_internet'] > 0
filter_b = dfsu['deficit_electricity'] > 0
dfsuinel = dfsu[(filter_a & filter_b)]
print(dfsuinel.shape)

(27, 27)


In [154]:
filter_r = dfsuinel['3_17enrolled_out'] < 1
dfsuinel_317 = dfsuinel[(filter_r)]
print(dfsuinel_317.shape)

(7, 27)


In [155]:
filter_c = dfsuinel['3_17enrolled_out'] > 0
dfsuinel317 = dfsuinel[(filter_c)]
print(dfsuinel317.shape)

(20, 27)


In [156]:
dfsuinel_317['id'].unique()

array([6019544148869120, 6510579757088768, 6220331101716480,
       5533645438976000, 5890664461238272, 4870399744802816,
       5390048441925632])

In [157]:
dfsuinel317['id'].unique()

array([5318023971078144, 6701378797043712, 6180819550076928,
       6725185465221120, 5906772132102144, 6071320583602176,
       5504083044859904, 6174731920474112, 6630508414107648,
       5147129168527360, 5750071173054464, 4618979254730752,
       6311683193831424, 4573554925633536, 5151084564971520,
       6733101559709696, 5240094356406272, 6386454480814080,
       6339256279891968, 5074428542320640])

#### Sucre, school/teachers internet availability (School view)

In [158]:
values = dfsu['st_internet'].value_counts()
labels = dfsu['st_internet'].value_counts().index

fig = px.pie(dfsu, names=labels, values=values, color_discrete_sequence=['lightpink','lightgreen',])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Carabobo

In [159]:
dfca = df[df['state'].isin(['Carabobo'])]
print(dfca.shape)

(161, 27)


In [160]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0.4,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Carabobo as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [161]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfca,
        orientation="h",
        title="Finance",
        x = dfca['finance'].value_counts(),
        y = dfca['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [162]:
dfca['finance'].value_counts()

I can afford food, but nothing else                                                49
I can afford food and regular expenses, but nothing else                           42
I cannot afford enough food for my family                                          35
I can afford food, regular expenses, and clothes, but nothing else                 11
Prefer not to answer                                                               10
I can comfortably afford food, clothes, and furniture, but I don’t have savings     8
I can comfortably afford food, clothes, and furniture, and I have savings           6
Name: finance, dtype: int64

In [163]:
dfca['age'].value_counts()

16 to 25 years old    60
26 to 35 years old    47
36 to 45 years old    32
Over 45 years old     22
Name: age, dtype: int64

#### Carabobo, deficit electric & internet, and enrolled out of school (Student view)

In [164]:
filter_a = dfca['deficit_internet'] > 0
filter_b = dfca['deficit_electricity'] > 0
dfcainel = dfca[(filter_a & filter_b)]
print(dfcainel.shape)

(51, 27)


In [165]:
filter_q = dfcainel['3_17enrolled_out'] < 1
dfcainel_317 = dfcainel[(filter_q)]
print(dfcainel_317.shape)

(13, 27)


In [166]:
filter_c = dfcainel['3_17enrolled_out'] > 0
dfcainel317 = dfcainel[(filter_c)]
print(dfcainel317.shape)

(38, 27)


In [167]:
dfcainel_317['id'].unique()

array([5846633379266560, 5250740372373504, 6384411083014144,
       6563523500179456, 5955565980483584, 5646365043720192,
       4759748351361024, 5129350570835968, 4951462286458880,
       6338158647312384, 5631680735084544, 5889487304654848,
       6325556005306368])

In [168]:
dfcainel317['id'].unique()

array([5203047025475584, 6198197138751488, 6606099762642944,
       4865311349407744, 5703385767739392, 5453711601303552,
       6373576491139072, 6502740116373504, 4753009497341952,
       4642877811720192, 5572631058448384, 6437927113719808,
       4814252643975168, 5171741898047488, 6541362207129600,
       4802723995451392, 5154074466189312, 6152833761280000,
       5658879974440960, 6221729935654912, 6028214949642240,
       4675851013849088, 5616931095969792, 6274833582391296,
       6458308612587520, 5114164103348224, 6390127365718016,
       4662930812698624, 6393447694204928, 6244017041047552,
       5577089771372544, 6370517518123008, 4622947670294528,
       4711810946301952, 5143940759289856, 4674185220849664,
       5382643968638976, 4714097949278208])

#### Carabobo, school/teachers internet availability (School view)

In [169]:
values = dfca['st_internet'].value_counts()
labels = dfca['st_internet'].value_counts().index

fig = px.pie(dfca, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

# States comparison

In [170]:
fig = px.bar(df,
             x="state",
             y="deficit_internet",
             title="State with deficit internet report at home"
              )
fig.update_layout(barmode="stack", xaxis={'categoryorder':'total descending'})
fig.show()

In [171]:
filter = df['deficit_internet'] > 0
dfdefinternet = df[filter]
dfdefinternet['state'].value_counts()

Miranda             414
Zulia               362
Distrito Federal    304
Táchira             257
Lara                215
Anzoátegui          191
Bolívar             145
Aragua              107
Carabobo             97
Sucre                92
Mérida               54
Falcón               45
Portuguesa           44
Guárico              42
Monagas              37
Trujillo             28
Yaracuy              26
Nueva Esparta        25
Apure                24
Vargas               21
Barinas              17
Cojedes              12
Delta Amacuro         8
Amazonas              2
Name: state, dtype: int64

In [172]:
fig = px.bar(df,
             x="state",
             y="deficit_electricity",
             title="State with deficit electricity report at home"
              )
fig.update_layout(barmode="stack", xaxis={'categoryorder':'total descending'})
fig.show()

In [173]:
filter = df['deficit_electricity'] > 0
dfdefelectric = df[filter]
dfdefelectric['state'].value_counts()

Táchira             219
Zulia               200
Miranda             153
Lara                112
Distrito Federal    105
Anzoátegui           64
Carabobo             59
Bolívar              58
Aragua               51
Mérida               43
Sucre                37
Falcón               27
Portuguesa           22
Guárico              21
Trujillo             19
Barinas              14
Monagas              13
Nueva Esparta        12
Apure                 8
Yaracuy               8
Vargas                7
Cojedes               6
Delta Amacuro         3
Amazonas              1
Name: state, dtype: int64

In [174]:
fig = px.bar(df,
             x="state",
             y="3_17enrolled_out",
             title="Children 3-17 years old enrolled and didn't return to school report"
              )
fig.update_layout(barmode="stack", xaxis={'categoryorder':'total descending'})
fig.show()

In [175]:
filter = df['3_17enrolled_out'] > 0
df317enrolled = df[filter]
df317enrolled['state'].value_counts()

Miranda             431
Zulia               369
Distrito Federal    337
Táchira             269
Lara                238
Anzoátegui          221
Bolívar             166
Carabobo            103
Sucre               101
Aragua               96
Mérida               53
Falcón               43
Guárico              41
Monagas              38
Portuguesa           35
Trujillo             34
Apure                31
Yaracuy              29
Vargas               27
Nueva Esparta        23
Barinas              21
Cojedes               7
Delta Amacuro         6
Amazonas              1
Name: state, dtype: int64

#Hidden states

## State Mérida

In [176]:
dfme = df[df['state'].isin(['Mérida'])]
print(dfme.shape)

(91, 27)


In [177]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0.3, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Mérida as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [178]:
values = dfme['age'].value_counts()
labels = dfme['age'].value_counts().index

fig = px.pie(dfme, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Age Distribution', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

#### Mérida, deficit electric & internet, and enrolled out of school (Student view)

In [179]:
filter_a = dfme['deficit_internet'] > 0
filter_b = dfme['deficit_electricity'] > 0
dfmeinel = dfme[(filter_a & filter_b)]
print(dfmeinel.shape)

(37, 27)


In [180]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfmeinel,
        orientation="h",
        title="Finance",
        x = dfmeinel['finance'].value_counts(),
        y = dfmeinel['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [181]:
filter_c = dfmeinel['3_17enrolled_out'] > 0
dfmeinel317 = dfmeinel[(filter_c)]
print(dfmeinel317.shape)

(22, 27)


In [182]:
dfmeinel317['id'].unique()

array([4593000155185152, 5223783077249024, 5746694389235712,
       5842444997623808, 5416570921943040, 6111522081472512,
       6052102752436224, 4765140682801152, 6291673998163968,
       5467730777210880, 5824073207906304, 4871789300940800,
       5511352260231168, 4817913549946880, 4917903089729536,
       5722066107236352, 5358371917266944, 4838725703172096,
       4725673238462464, 5501215600541696, 5497123385901056,
       6659254948986880])

#### Mérida, school/teachers internet availability (School view)

In [183]:
values = dfme['st_internet'].value_counts()
labels = dfme['st_internet'].value_counts().index

fig = px.pie(dfme, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Falcón

In [184]:
dffa = df[df['state'].isin(['Falcón'])]
print(dffa.shape)

(67, 27)


In [185]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0.3, 0, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Falcón as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [186]:
values = dffa['age'].value_counts()
labels = dffa['age'].value_counts().index

fig = px.pie(dffa, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Age Distribution', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

#### Falcón, deficit electric & internet, and enrolled out of school (Student view)

In [187]:
filter_a = dffa['deficit_internet'] > 0
filter_b = dffa['deficit_electricity'] > 0
dffainel = dffa[(filter_a & filter_b)]
print(dffainel.shape)

(25, 27)


In [188]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dffainel,
        orientation="h",
        title="Finance",
        x = dffainel['finance'].value_counts(),
        y = dffainel['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [189]:
filter_c = dffainel['3_17enrolled_out'] > 0
dffainel317 = dffainel[(filter_c)]
print(dffainel317.shape)

(15, 27)


In [190]:
dffainel317['id'].unique()

array([5887497107406848, 6192312144363520, 6400688774184960,
       6397857417658368, 5265153879965696, 6140627531333632,
       6555868593389568, 5582189038403584, 5320852005715968,
       6279988415561728, 5978184251342848, 5880129762099200,
       5571436537446400, 4620943883501568, 5251079540572160])

#### Falcón, school/teachers internet availability (School view)

In [191]:
values = dffa['st_internet'].value_counts()
labels = dffa['st_internet'].value_counts().index

fig = px.pie(dffa, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Monagas

In [192]:
dfmo = df[df['state'].isin(['Monagas'])]
print(dfmo.shape)

(66, 27)


In [193]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0.3, 0, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Monagas as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [194]:
values = dfmo['age'].value_counts()
labels = dfmo['age'].value_counts().index

fig = px.pie(dfmo, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Age Distribution', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

#### Monagas, deficit electric & internet, and enrolled out of school (Student view)

In [195]:
filter_a = dfmo['deficit_internet'] > 0
filter_b = dfmo['deficit_electricity'] > 0
dfmoinel = dfmo[(filter_a & filter_b)]
print(dfmoinel.shape)

(11, 27)


In [196]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfmoinel,
        orientation="h",
        title="Finance",
        x = dfmoinel['finance'].value_counts(),
        y = dfmoinel['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [197]:
filter_c = dfmoinel['3_17enrolled_out'] > 0
dfmoinel317 = dfmoinel[(filter_c)]
print(dfmoinel317.shape)

(7, 27)


In [198]:
dfmoinel317['id'].unique()

array([4643105176551424, 5327093213954048, 5843205945032704,
       6378290821726208, 4862261855518720, 4677006091616256,
       4790614955917312])

#### Monagas, school/teachers internet availability (School view)

In [199]:
values = dfmo['st_internet'].value_counts()
labels = dfmo['st_internet'].value_counts().index

fig = px.pie(dfmo, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Guárico

In [200]:
dfgu = df[df['state'].isin(['Guárico'])]
print(dfgu.shape)

(58, 27)


In [201]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0.4, 0,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Guárico as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [202]:
values = dfgu['age'].value_counts()
labels = dfgu['age'].value_counts().index

fig = px.pie(dfgu, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Age Distribution', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

#### Guárico, deficit electric & internet, and enrolled out of school (Student view)

In [203]:
filter_a = dfgu['deficit_internet'] > 0
filter_b = dfgu['deficit_electricity'] > 0
dfguinel = dfgu[(filter_a & filter_b)]
print(dfguinel.shape)

(21, 27)


In [204]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfguinel,
        orientation="h",
        title="Finance",
        x = dfguinel['finance'].value_counts(),
        y = dfguinel['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [205]:
filter_c = dfguinel['3_17enrolled_out'] > 0
dfguinel317 = dfguinel[(filter_c)]
print(dfguinel317.shape)

(13, 27)


In [206]:
dfguinel317['id'].unique()

array([5614173240426496, 5505348701257728, 5311778283323392,
       6679550229741568, 4531028273397760, 5188244840120320,
       6052146306088960, 5992803279831040, 4714253910278144,
       5735505949097984, 6059452515221504, 4990786335932416,
       5454038085926912])

#### Guárico, school/teachers internet availability (School view)

In [207]:
values = dfgu['st_internet'].value_counts()
labels = dfgu['st_internet'].value_counts().index

fig = px.pie(dfgu, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Portuguesa

In [208]:
dfpo = df[df['state'].isin(['Portuguesa'])]
print(dfpo.shape)

(54, 27)


In [209]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0.3,
                                                                 0, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Portuguesa as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [210]:
values = dfpo['age'].value_counts()
labels = dfpo['age'].value_counts().index

fig = px.pie(dfpo, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Age Distribution', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

#### Portuguesa, deficit electric & internet, and enrolled out of school (Student view)

In [211]:
filter_a = dfpo['deficit_internet'] > 0
filter_b = dfpo['deficit_electricity'] > 0
dfpoinel = dfpo[(filter_a & filter_b)]
print(dfpoinel.shape)

(19, 27)


In [212]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfpoinel,
        orientation="h",
        title="Finance",
        x = dfpoinel['finance'].value_counts(),
        y = dfpoinel['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [213]:
filter_c = dfpoinel['3_17enrolled_out'] > 0
dfpoinel317 = dfpoinel[(filter_c)]
print(dfpoinel317.shape)

(14, 27)


In [214]:
dfpoinel317['id'].unique()

array([6668286678007808, 5663870927765504, 5690145524416512,
       4888339605094400, 4977557165572096, 6155877311840256,
       5349780288176128, 5501189109317632, 4506570515021824,
       6058875580317696, 5971985573347328, 6048474713030656,
       4810534527696896, 4739388948086784])

#### Portuguesa, school/teachers internet availability (School view)

In [215]:
values = dfpo['st_internet'].value_counts()
labels = dfpo['st_internet'].value_counts().index

fig = px.pie(dfpo, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Apure

In [216]:
dfap = df[df['state'].isin(['Apure'])]
print(dfap.shape)

(54, 27)


In [217]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0.3, 0, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Apure as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [218]:
values = dfap['age'].value_counts()
labels = dfap['age'].value_counts().index

fig = px.pie(dfap, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Age Distribution', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

#### Apure, deficit electric & internet, and enrolled out of school (Student view)

In [219]:
filter_a = dfap['deficit_internet'] > 0
filter_b = dfap['deficit_electricity'] > 0
dfapinel = dfap[(filter_a & filter_b)]
print(dfapinel.shape)

(7, 27)


In [220]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfapinel,
        orientation="h",
        title="Finance",
        x = dfapinel['finance'].value_counts(),
        y = dfapinel['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [221]:
filter_c = dfapinel['3_17enrolled_out'] > 0
dfapinel317 = dfapinel[(filter_c)]
print(dfapinel317.shape)

(5, 27)


In [222]:
dfapinel317['id'].unique()

array([5585967133229056, 6299149522960384, 6568629478096896,
       4543119076098048, 6289968174727168])

#### Apure, school/teachers internet availability (School view)

In [223]:
values = dfap['st_internet'].value_counts()
labels = dfap['st_internet'].value_counts().index

fig = px.pie(dfap, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

## State Trujillo

In [224]:
dftr = df[df['state'].isin(['Trujillo'])]
print(dftr.shape)

(53, 27)


In [225]:
values = df['state'].value_counts()
labels = df['state'].value_counts().index

# pull is given as a fraction of the pie radius
fig = go.Figure(data=[go.Pie(labels=labels, values=values, pull=[0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0, 0, 0, 0,
                                                                 0, 0.3, 0, 0, 0, 
                                                                 0, 0, 0, 0])])
fig.update_layout(title='Trujillo as submission source was ____ of the data', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [226]:
values = dfap['age'].value_counts()
labels = dfap['age'].value_counts().index

fig = px.pie(dfap, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Age Distribution', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

#### Apure, deficit electric & internet, and enrolled out of school (Student view)

In [227]:
filter_a = dfap['deficit_internet'] > 0
filter_b = dfap['deficit_electricity'] > 0
dfapinel = dfap[(filter_a & filter_b)]
print(dfapinel.shape)

(7, 27)


In [228]:
colors = ['lightslategray',] * 5

fig = px.bar(
        dfapinel,
        orientation="h",
        title="Finance",
        x = dfapinel['finance'].value_counts(),
        y = dfapinel['finance'].value_counts().index,
        template='plotly_white'
        )
fig.update_layout(barmode='stack', yaxis={'categoryorder':'total ascending'})
fig.update_traces(marker_color='green', marker_line_color='#111c2d',
                  marker_line_width=1.5, opacity=0.6)
fig.show()

In [229]:
filter_c = dfapinel['3_17enrolled_out'] > 0
dfapinel317 = dfapinel[(filter_c)]
print(dfapinel317.shape)

(5, 27)


In [230]:
dfapinel317['id'].unique()

array([5585967133229056, 6299149522960384, 6568629478096896,
       4543119076098048, 6289968174727168])

#### Apure, school/teachers internet availability (School view)

In [231]:
values = dfap['st_internet'].value_counts()
labels = dfap['st_internet'].value_counts().index

fig = px.pie(dfap, names=labels, values=values, color_discrete_sequence=['lightgreen', 'lightpink'])
fig.update_layout(title='Does school/teachers have internet connection? 0=No, 1=Yes', template='plotly_dark', hoverlabel=dict(font_size=45))
fig.update_traces(textposition='inside', textinfo='percent+label+value')
fig.show()

# Coba

In [232]:
## Coba

In [233]:
values = df['region'].value_counts()
labels = df['region'].value_counts().index

fig = px.pie(df, names=labels, values=values)

for template in ["plotly", "plotly_white", "plotly_dark", "ggplot2", "seaborn", "simple_white", "none"]:
  fig.update_layout(template=template, hoverlabel=dict(font_size=20), title="Region Distribution: '%s' theme" % template)
  fig.update_traces(textposition='inside', textinfo='percent+label+value')
  fig.show()

In [234]:
fig = px.bar(
        df,
        orientation="h",
        title="Education",
        x = df['education'].value_counts(),
        y = df['education'].value_counts().index,
        )

for template in ["plotly", "plotly_white", "plotly_dark", "ggplot2", "seaborn", "simple_white", "none"]:
  fig.update_layout(template=template, hoverlabel=dict(font_size=20), title="Education Distribution: '%s' theme" % template,
                    barmode='stack', yaxis={'categoryorder':'total ascending'})

  fig.show()